cache_control.go

```diff
@@ -12,18 +13,26 @@ import (
 // underlying server's method implementation to allow control over the
 // duration and nature of caching on a per-request basis.
 type CacheControl struct {
+	sync.RWMutex
+
 	// MaxAge is maximum duration (since the original retrieval) that
 	// an item is considered fresh.
 	MaxAge time.Duration
 }
 
 func (cc *CacheControl) cacheable() bool {
-	return cc.MaxAge > 0
+	cc.RLock()
```

Comment:
`defer cc.RUnlock()` is more idiomatic, I think, and more robust to later changes that might be made to the code.